### PR TITLE
[ENGG-2169] feat: auto upgrade plan

### PR DESCRIPTION
**Pull request by nafees87n**

### CODE REVIEW

File: app/src/features/pricing/components/UpgradeSuccess.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  const navigate = useNavigate();
+  const user = useSelector(getUserAuthDetails);
+  const userPlanName = user?.details?.planDetails?.planName;
+
+  useEffect(() => {
+    if (user.loggedIn && userPlanName) {
+      const sendPlanUpgradedNotification = httpsCallable(
+        getFunctions(),
+        "premiumNotifications-sendPlanUpgradedNotification"
+      );
```
Comment (nsrCodes) [inline app/src/features/pricing/components/UpgradeSuccess.tsx:23]: can be moved outside the useEffect

File: app/src/features/pricing/components/UpgradeSuccess.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      sendPlanUpgradedNotification({
+        planName: getPrettyPlanName(userPlanName),
+      }).catch((e) => {
+        Logger.log(e);
```
Comment (nsrCodes) [inline app/src/features/pricing/components/UpgradeSuccess.tsx:27]: use Logger.error instead?

File: app/src/features/pricing/components/UpgradeSuccess.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      });
+
+      getBillingTeamRedirectURL(user?.details?.profile?.uid).then((redirectUrl) => {
+        if (!redirectUrl) {
+          navigate(PATHS.SETTINGS.BILLING.RELATIVE);
+        } else {
+          navigate(redirectUrl);
```
Comment (nsrCodes) [inline app/src/features/pricing/components/UpgradeSuccess.tsx:34]: do we need to add some extra params in the redirecUrl like [here](https://github.com/requestly/requestly/blob/ENGG-2169/app/src/features/pricing/components/PricingModal/Checkout.tsx#L46)

**Review by nsrCodes — APPROVED** [review]

File: app/src/features/pricing/components/UpgradeSuccess.tsx (hunk 3 above)
Comment (nafees87n) [reply app/src/features/pricing/components/UpgradeSuccess.tsx:34]: No, not needed

**Pull request merged**